testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/configurationcache/ConfigurationCacheProblemsFixture.groovy

```diff
@@ -45,234 +49,127 @@ import static org.hamcrest.CoreMatchers.notNullValue
 import static org.hamcrest.CoreMatchers.nullValue
 import static org.hamcrest.CoreMatchers.startsWith
 import static org.hamcrest.MatcherAssert.assertThat
-import static org.junit.Assert.assertThrows
 import static org.junit.Assert.assertTrue
 
-final class ConfigurationCacheProblemsFixture {
-
+class ConfigurationCacheProblemsFixture {
     protected static final String PROBLEMS_REPORT_HTML_FILE_NAME = "configuration-cache-report.html"
 
-    private final File rootDir
+    protected final TestFile rootDir
 
     ConfigurationCacheProblemsFixture(File rootDir) {
-        this.rootDir = rootDir
-    }
-
-    void assertFailureHasError(
-        ExecutionFailure failure,
-        String error,
-        @DelegatesTo(value = HasConfigurationCacheErrorSpec, strategy = Closure.DELEGATE_FIRST) Closure<?> specClosure
-    ) {
-        assertFailureHasError(failure, error, ConfigureUtil.configureUsing(specClosure))
+        this.rootDir = rootDir instanceof TestFile ? rootDir : new TestFile(rootDir)
```

Comment:
wrapping plain `File` objects as `TestFile` as a convenience here, instead of requiring call sites to do it